services/codacy/codacy-grade.tester.js

```diff
@@ -0,0 +1,45 @@
+'use strict'
+
+const Joi = require('joi')
+const { codacyGrade } = require('./codacy-helpers')
+
+const t = require('../create-service-tester')()
+module.exports = t
+
+t.create('Code quality')
+  .get('/grade/e27821fb6289410b8f58338c7e0bc686.json')
+  .expectJSONTypes(
+    Joi.object().keys({
+      name: 'code quality',
+      value: codacyGrade,
+    })
+  )
+
+t.create('Code quality on branch')
+  .get('/grade/e27821fb6289410b8f58338c7e0bc686/master.json')
+  .expectJSONTypes(
+    Joi.object().keys({
+      name: 'code quality',
+      value: codacyGrade,
+    })
+  )
+
+t.create('Code quality (package not found)')
+  .get('/grade/00000000000000000000000000000000/master.json')
+  .expectJSON({
+    name: 'code quality',
+    value: 'project or branch not found',
+  })
+
+// This is a known bug. The badge endpoint for a nonexistent branch returns
+// the same result. It seems possible the branch specification isn't being
+// considered at all.
+// e.g.
+// https://api.codacy.com/project/badge/grade/e27821fb6289410b8f58338c7e0bc686
+// https://api.codacy.com/project/badge/grade/e27821fb6289410b8f58338c7e0bc686?branch=foo
+// t.create('Code quality on branch (branch not found)')
```

Comment:
We've been following the [documentation](https://support.codacy.com/hc/en-us/articles/212799365-Badges) regarding the branch parameter, either it's out of date or the feature isn't working properly. @nemchik has already contacted the support team in the past, maybe it's worth him enquiring about this parameter? 😄

Reply:
Ah, right. I reached out to them on Twitter a few days back. Their response was the branch parameter is used if it matches a branch, and it's ignored if it doesn't match a branch, i.e. it fails silently. I'll update the comment.